### PR TITLE
clang-tidy fixes pt.5 

### PR DESCRIPTION
**Pull request by rdspring1**

### CODE REVIEW

File: csrc/register_interface.cpp [hunk 1]
```diff
@@ -607,7 +610,7 @@ torch::jit::RegisterOperators reg_infer_index_select({
             auto idx_size = torch::jit::pop(stack).toIntVector();
             auto size = torch::jit::pop(stack).toIntVector();
             size[selected_dim] = idx_size[0];
-            torch::jit::push(stack, c10::IValue(std::move(size)));
+            torch::jit::push(stack, c10::IValue(size));
```
Comment (zasdfgbnm) [inline csrc/register_interface.cpp:613]: Why `std::move` is removed?

File: csrc/sys_utils.cpp [hunk 2]
```diff
@@ -29,19 +30,19 @@ std::string disassembleBinary(
   // Reference:
   // https://stackoverflow.com/a/3469651
   // https://linuxhint.com/dup2_system_call_c/
-  pid_t pid;
```
Comment (zasdfgbnm) [inline csrc/sys_utils.cpp:32]: I don't think you can `for` here. It must happen after pipe.

File: csrc/transform_replay.cpp [hunk 3]
```diff
@@ -989,7 +989,7 @@ bool validateDomain(TensorView* tv, TensorDomain* new_td) {
 } // namespace
 
 void TransformPropagator::propagateC2P(TensorView* from, TensorView* to) {
-  int pos = replayed_pos_.at(from);
+  int pos = (int)replayed_pos_.at(from);
```
Comment (naoyam) [inline csrc/transform_replay.cpp:992]: Shouldn't we make `replayed_pos_` to hold `int` rather than `size_t`? @zasdfgbnm

Comment (zasdfgbnm) [reply csrc/transform_replay.cpp:992]: I agree. I think this makes more sense.

File: csrc/register_interface.cpp (hunk 1 above)
Comment (rdspring1) [reply csrc/register_interface.cpp:613]: Error (CLANGTIDY) [performance-move-const-arg,-warnings-as-errors] passing result of std::move() as a const reference argument; no move will actually happen

        607  |            auto idx_size = torch::jit::pop(stack).toIntVector();
        608  |            auto size = torch::jit::pop(stack).toIntVector();
        609  |            size[selected_dim] = idx_size[0];
    >>> 610  |            torch::jit::push(stack, c10::IValue(std::move(size)));
        611  |          };
        612  |        },
        613  |        torch::jit::aliasAnalysisFromSchema()),

**Comment by naoyam** [comment]
LGTM. I'll let Xiang to approve the PR.

File: csrc/transform_replay.cpp [hunk 4]
```diff
@@ -1088,20 +1088,20 @@ void TransformPropagator::propagateSibling(TensorView* from, TensorView* to) {
   replayed_pos_[to] = pos;
 }
 
-TransformPropagator::TransformPropagator(TensorView* from, int64_t pos) {
+TransformPropagator::TransformPropagator(TensorView* from, int pos) {
```
Comment (zasdfgbnm) [inline csrc/transform_replay.cpp:1091]: Why are we changing this to `int`? Shouldn't we use `int64_t` everywhere according to https://github.com/NVIDIA/Fuser/pull/82#discussion_r1150913183?

**Review by zasdfgbnm — APPROVED** [review]
The `fork` issue has been resolved. Regarding the handling `int` vs `int64_t`, there might be a more consistent way to do it, but I strongly want to see clang-tidy enabled ASAP, rather than having perfect integer handling, so feel free to merge as is.

File: csrc/transform_replay.cpp (hunk 4 above)
Comment (rdspring1) [reply csrc/transform_replay.cpp]: I updated the functions in `transform_replay.cpp` to use `int64_t`

**Pull request merged**